### PR TITLE
ci: migrate to goreleaser

### PR DESCRIPTION
**Pull request by burntcarrot**
## Related Issue
Closes: #117

#### Describe the changes you've made
This PR introduces making releases using GoReleaser. The web UI is fetched from `keploy/ui` and is embedded into the binary while creating a new release.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by burntcarrot** [comment]
Here is a preview (test release): https://github.com/burntcarrot/keploy/releases/tag/v0.3.4

File: goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# GoReleaser configuration
+
+before:
+  hooks:
+  - go mod tidy
```
Comment (slayerjain) [inline goreleaser.yaml:5]: I think it would be best to avoid this because it could update versions.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,29 +1,25 @@
-name: Go
-
+name: Release
 on:
   push:
-    branches: [ main ]
```
Comment (slayerjain) [inline .github/workflows/release.yml:5]: also, I think we should move the contents of this file to another file. This file was meant to run whenever any commit was pushed to master.

**Review by slayerjain — CHANGES_REQUESTED** [review]
@burntcarrot Thanks for the PR. It looks awesome except for just minor changes. maybe you could just move the existing contents of `release.yml` to another file called `main.yml` which would run on every commit pushed to master.

**Review by slayerjain — APPROVED** [review]

**Comment by slayerjain** [comment]
@burntcarrot This look awesome, Thanks! :)

**Pull request merged**